### PR TITLE
Upgrade to Storybook 8

### LINKED ISSUE #231: Storybook 8 upgrade
**Is your feature request related to a problem? Please describe.**
When starting up storybook 8 it states this Addon is out of date.

**Describe the solution you'd like**
It still works though so I'm guessing just need a version release to appease storybook?

**Describe alternatives you've considered**

**Design types**

- [ ] `figma`
- [ ] `figspec`
- [ ] `link`
- [ ] `image`
- [ ] `iframe`

**Additional context**

This is the warning I get.
image

Your addon is not listed on https://github.com/storybookjs/storybook/issues/26031 though.

### PR DESCRIPTION
**Pull request by JReinhold**
Closes #231 

This PR does a lot of things:

1. Upgrade bundling strategy to use tsup similar to how it's done in addon-kit
2. Upgrade all internal Storybooks to SB 8
3. Combine `examples` and `examples-tab` Storybooks into a single Storybook, while still maintaining their separation of `/` for examples and `/tab/` for tab examples
4. Upgrade almost all dependencies throughout the repository
5. Formatted the project according to the existing configuration

## Todo

- [x] Fix the link in docs between panel examples and tab examples
- [x] Fix the table in the docs

<details>
  <summary>📦 Published PR as canary version: <code>8.0.0--canary.232.a463deb.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-designs@8.0.0--canary.232.a463deb.0
  # or 
  yarn add @storybook/addon-designs@8.0.0--canary.232.a463deb.0
  ```
</details>

### CODE REVIEW

File: packages/storybook-addon-designs/package.json [hunk 1]
```diff
@@ -2,83 +2,108 @@
...
-    "@storybook/preview-api": "^8.0.0-alpha.10 || ^8.0.0",
-    "@storybook/theming": "^8.0.0-alpha.10 || ^8.0.0",
-    "react": "^16.8.0 || ^17.0.0 || ^18.0.0",
-    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0"
+    "@storybook/blocks": "^8.0.0",
+    "@storybook/components": "^8.0.0",
+    "@storybook/theming": "^8.0.0",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
   },
```
Comment (JReinhold) [inline packages/storybook-addon-designs/package.json:69]: These are only needed by users if they intend to use the doc blocks this package provides.

**Comment by valentinpalkovic** [comment]
@JReinhold: Out of the 13 reported issues, how many would you consider helpful and valuable? And a second question: Most errors might not have been reported if TypeScript had been set up stricter (strict mode, for example)

**Comment by JReinhold** [comment]
@valentinpalkovic quickly looking through them, they would all have been caught by a stricter TypeScript mode, so I don't find this specific scenario useful, except for anything else than "you should make your TS config more strict".

**Comment by bodograumann** [comment]
I have tried 8.0.0--canary.232.785af00.0 and it works fine afaics. :heart:

**Review by JonathanKolnik — APPROVED** [review]
👍 👍

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 <div align="center">
   
-  <img src="./packages/assets/logo.png" width="104" alt="logo">
+  <img src="./packages/examples/assets/logo.png" width="104" alt="logo">
   <br/>
   <br/>
```
[the remark this answers is not in the document]
Comment (JReinhold) [reply README.md:6]: Good catch, I updated the required Storybook version, with a note about v7 supporting SB 7.

**Pull request merged**